### PR TITLE
Incremental conversion of codebase to es6 classes - api

### PR DESCRIPTION
**Pull request by scalvert**
Currently the codebase is largely utilizing es5 objects w/prototypical inheritance. This PR starts the conversion to es6 classes to help improve code cleanliness and readability.

Converts api.js.

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -48,61 +48,61 @@ var Server = require('./server');
...
-  this.configureLogging();
-  log.info('Test\'em starting..');
-  this.config.read(function() {
-    self.app = new App(self.config, finalizer);
-    self.app.start();
-  });
-};
+class Api {
+  setup(mode, finalizer) {
+    var self = this;
```
Comment (stefanpenner) [inline lib/api.js]: this `self` can be removed, if the bellow `self` becomes `this`.

File: lib/api.js [hunk 2]
```diff
@@ -48,61 +48,61 @@ var Server = require('./server');
...
+  startServer(options) {
+    this.options = options;
+    this.config = new Config('server', this.options);
+    this.config.setDefaultOptions(this.defaultOptions);
+    var config = this.config;
+    config.read(() => {
+      var server = new Server(config);
+      server.start();
+      server.on('server-start', () => {
+        console.log('Open ' + config.get('url') + ' in a browser to connect.');
```
Comment (stefanpenner) [inline lib/api.js]: `this.config`

File: lib/api.js [hunk 3]
```diff
@@ -48,61 +48,61 @@ var Server = require('./server');
...
-  config.read(function() {
-    var server = new Server(config);
-    server.start();
-    server.on('server-start', function() {
-      console.log('Open ' + config.get('url') + ' in a browser to connect.');
+  startServer(options) {
+    this.options = options;
+    this.config = new Config('server', this.options);
+    this.config.setDefaultOptions(this.defaultOptions);
+    var config = this.config;
```
Comment (stefanpenner) [inline lib/api.js]: this can be dropped

**Comment by scalvert** [comment]
Fixed the selfs, @stefanpenner

**Pull request merged**